### PR TITLE
shared update layer handler

### PR DESCRIPTION
**Pull request by agathanatasha**
The update_layer function is repeated across many charms. This PR adds this function to chisme to be shared across all charms in the future.

### CODE REVIEW

File: src/charmed_kubeflow_chisme/pebble/_update_layer.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+from logging import Logger
+
+from ops.model import BlockedStatus, Container
+from ops.pebble import ChangeError, Layer
+
+from ..exceptions import ErrorWithStatus
+
+
+def update_layer(container_name: str, container: Container, new_layer: Layer, logger: Logger):
+    """Updates the Pebble configuration layer if changed."""
```
Comment (DnPlas) [inline src/charmed_kubeflow_chisme/pebble/_update_layer.py:14]: I'd add some docstrings with an `Args` section and what this method raises on failure.

**Review by DnPlas — COMMENTED** [review]
Thanks @agathanatasha , left some comments.

File: tests/test_pebble.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    def side_effect():
+        raise ChangeError("error message", "changelog")
+
+    mocked_container.replan.side_effect = side_effect
+
+    new_layer = Layer({"services": {container_name: {"summary": "new plan"}}})
+
+    with pytest.raises(ErrorWithStatus):
+        update_layer(container_name, mocked_container, new_layer, mocked_logger)
```
Comment (DnPlas) [inline tests/test_pebble.py:68]: I'd add one more test case for checking the service is actually running. Similar to [this](https://github.com/canonical/template-operator/blob/main/tests/unit/test_charm.py#L65)

**Review by DnPlas — COMMENTED** [review]
One more comment, other than that it LGTM

File: tests/test_pebble.py (hunk 2 above)
Comment (agathanatasha) [reply tests/test_pebble.py:68]: The whole container object is mocked and the tests are not setup with harness. `is_running` would also be mocked out. I don't think that would be useful

**Review by DnPlas — APPROVED** [review]
Thanks @agathanatasha

**Pull request merged**